Owl/app/src/main/java/com/example/owl/ui/MainActivity.kt

```diff
@@ -33,6 +34,7 @@ class MainActivity : ComponentActivity() {
             )
         )
         super.onCreate(savedInstanceState)
+        ViewCompat.setOnApplyWindowInsetsListener(window.decorView) { _, insets -> insets }
```

Comment:
This gets the inset and return it without doing anything. It seems to me that it doesn't do what the before/after screenshots do.
Is the PR missing some commits?

Reply:
No, it isn't.
The MDC Catalog app does the same thing for edge-to-edge layout.
https://github.com/material-components/material-components-android/blob/master/catalog/java/io/material/catalog/windowpreferences/WindowPreferencesManager.java#L80-L81
https://github.com/material-components/material-components-android/blob/master/catalog/java/io/material/catalog/windowpreferences/WindowPreferencesManager.java#L41-L46

Reply:
In portrait orientation, the listener in MDC Catalog app only returns the passed parameter, too. This causes that it is drawn behind the system navigation bar in 2 & 3-button navigation mode.

Reply:
I see, so this is for disabling the contrast enforcement for the 3-button navbar. That is a side effect of preventing the original `onApplyWindowInsets` of the `decorView`.
https://cs.android.com/android/platform/superproject/main/+/main:frameworks/base/core/java/com/android/internal/policy/DecorView.java;l=1030
The problem is that it also prevents other things done in that method. For example, `R.color.immersive_sys_ui` set above won't be used.

If you want the 3-button nav to be transparent, you should use the `navigationBarStyle` parameter in `enableEdgeToEdge`.

Reply:
I used it.

Reply:
I don't understand that window.isNavigationBarContrastEnforced = true only when navigationBarStyle = SystemBarStyle.auto() (window.isNavigationBarContrastEnforced = false when navigationBarStyle = SystemBarStyle.dark() or SystemBarStyle.light()).
https://github.com/androidx/androidx/blob/androidx-main/activity/activity/src/main/java/androidx/activity/EdgeToEdge.kt#L315-L316

Reply:
The default appearance of 3-button navigation is to have a translucent scrim with `isNavigationBarContrastEnforced = true`. It is still considered edge-to-edge.
If you really want to make it transparent, you can do so, but that's optional.

Reply:
Thank you. But, I don't understand that window.isNavigationBarContrastEnforced = false when navigationBarStyle = SystemBarStyle.dark() or SystemBarStyle.light().

Does this PR still need fixing?

P.S.
Should we use the navigationBarStyle parameter in enableEdgeToEdge instead of ViewCompat.setOnApplyWindowInsetsListener() inside other apps (Jetchat, Jetsnack, and Reply)?

Reply:
`SystemBarStyle.auto` sets up with the recommended style (transparent gesture nav and translucent 3-button nav). The color parameters are for backward compatibility, and not suitable for color customization.

If you really want the bar to be transparent on 3-button nav, then you can use either `SystemBarStyle.light` or `SystemBarStyle.dark`. This app seems to change the background between the light/dark modes, so you need to call the right one based on the mode. I would leave this `enableEdgeToEdge` as is, and add a second call in `Themes.kt` because the mode `isDarkMode` is available there.

I don't recommend using `ViewCompat.setOnApplyWindowInsetsListener` for suppressing the DecorView's default behavior.

And the same goes for all the other PRs.

Reply:
Thank you. I tried writing it.

Reply:
https://github.com/android/compose-samples/pull/1265/commits/a1e7c597eb85cec716f5e83575a18b283278f6e2:
![Screenshot_20240220_150514](https://github.com/android/compose-samples/assets/71050561/c20a8ccc-aa99-4263-85c6-3b254a1a9ad2)